h/views/activity.py

```diff
@@ -90,27 +92,25 @@ def user_link(userid):
         }
 
 
-@view_defaults(route_name='activity.group_search',
-               renderer='h:templates/activity/search.html.jinja2')
+@view_defaults(route_name='group_read',
+               renderer='h:templates/activity/search.html.jinja2',
+               custom_predicates=[custom_predicates.has_read_permission,
+                                  custom_predicates.feature('search_page')])
 class GroupSearchController(SearchController):
-    """View callables unique to the "activity.group_search" route."""
+    """View callables unique to the "group_read" route."""
+
+    def __init__(self, group, request):
+        super(GroupSearchController, self).__init__(request)
+        self.group = group
+        check_slug(self.group, self.request)
```

Comment:
So all the view methods will redirect to the correct slug if it's wrong or missing